src/helm/proxy/clients/together_client.py

```diff
@@ -72,28 +87,84 @@ def __init__(self, cache_config: CacheConfig, api_key: Optional[str] = None):
         self.api_key: Optional[str] = api_key
         self.cache = Cache(cache_config)
 
+    def _get_job_url(self, job_id: str) -> str:
+        return f"https://api.together.xyz/jobs/job/{job_id}"
+
     def make_request(self, request: Request) -> RequestResult:
         raw_request = TogetherClient.convert_to_raw_request(request)
         cache_key: Dict = Client.make_cache_key(raw_request, request)
 
-        def do_it():
-            if not self.api_key:
-                raise TogetherClientError("togetherApiKey not set in credentials.conf")
-            headers: Dict[str, str] = {"Authorization": f"Bearer {self.api_key}"}
-            response = requests.post(TogetherClient.INFERENCE_ENDPOINT, headers=headers, json=raw_request)
-            try:
-                response.raise_for_status()
-            except Exception as e:
-                raise TogetherClientError(
-                    f"Together request failed with {response.status_code}: {response.text}"
-                ) from e
-            result = response.json()
-            return result["output"]
-
-        def fail():
-            raise RuntimeError(f"The result has not been uploaded to the cache for the following request: {cache_key}")
-
-        response, cached = self.cache.get(cache_key, wrap_request_time(do_it if self.api_key else fail))
+        if not self.api_key:
+            raise TogetherClientError("togetherApiKey not set in credentials.conf")
+        headers: Dict[str, str] = {"Authorization": f"Bearer {self.api_key}"}
+
+        if request.model_engine in _ASYNC_MODELS:
+
+            def submit_job() -> str:
+                submit_request = raw_request.copy()
+                submit_request["async"] = True
+                submit_response = requests.post(TogetherClient.INFERENCE_ENDPOINT, headers=headers, json=submit_request)
+                try:
+                    submit_response.raise_for_status()
+                except Exception as e:
+                    raise TogetherClientError(
+                        f"Together job submission request failed with {submit_response.status_code}: "
+                        f"{submit_response.text}"
+                    ) from e
+                submit_response_json = submit_response.json()
+                job_id = submit_response_json.get("id")
+                if not job_id:
+                    raise TogetherClientError(
+                        f"Could not get job_id from job submission response {submit_response_json}"
+                    )
+                return job_id
+
+            # Retry with a 1 second delay that doubles every retry until a maximum of delay of 15 seconds.
+            # Stop retrying after 1 minute.
+            @retry(wait_exponential_multiplier=1000, wait_exponential_max=15000, stop_max_delay=60000)
```

Comment:
Does the retry respond to any Exception?  Could we be more selective about this to avoid swallowing up exceptions unintentionally?

Reply:
Made it only retry `JobNotFinishedError`.